opensearch/src/main/java/org/opensearch/sql/opensearch/response/OpenSearchResponse.java

```diff
@@ -92,23 +95,35 @@ public Iterator<ExprValue> iterator() {
         return (ExprValue) ExprTupleValue.fromExprValueMap(builder.build());
       }).iterator();
     } else {
+      ExprFloatValue maxScore = Float.isNaN(hits.getMaxScore())
+          ? null : new ExprFloatValue(hits.getMaxScore());
       return Arrays.stream(hits.getHits())
           .map(hit -> {
-            ExprValue docData = exprValueFactory.construct(hit.getSourceAsString());
-            if (hit.getHighlightFields().isEmpty()) {
-              return docData;
-            } else {
-              ImmutableMap.Builder<String, ExprValue> builder = new ImmutableMap.Builder<>();
-              builder.putAll(docData.tupleValue());
+            String source = hit.getSourceAsString();
+            ExprValue docData = exprValueFactory.construct(source);
+
+            ImmutableMap.Builder<String, ExprValue> builder = new ImmutableMap.Builder<>();
+            builder.putAll(docData.tupleValue());
+            builder.put("_index", new ExprStringValue(hit.getIndex()));
+            builder.put("_id", new ExprStringValue(hit.getId()));
+            if (!Float.isNaN(hit.getScore())) {
+              builder.put("_score", new ExprFloatValue(hit.getScore()));
+            }
+            if (maxScore != null) {
+              builder.put("_maxscore", maxScore);
+            }
+            builder.put("_sort", new ExprLongValue(hit.getSeqNo()));
```

Comment:
Any chance we only add fields pushed down to TableScan operator to response? Otherwise we always populate all meta field values to each result row.